### PR TITLE
feat(subgraph): add subgraph module for indexing data

### PR DESCRIPTION
**Pull request by hangleang**
# Description

Add subgraph module and integrate into the workspace with the following functionality:

- support multiple network's deployment
- automate saved deployment into `networks.json` file for generating `subgraph.yaml`
- index all `Poll` related events, which has been deployed from the `MACI` contract instance

## Additional Notes

**every deployment of MACI contracts will be automatically updated the `networks.json` file, but you need to manually generate the `subgraph.yaml` by running `pnpm run codegen:local` in the subgraph directory**

There is also some limitation of current implementation including:
- missing some properties to serve as complete data layer for MACI due to only indexing the events, not yet include the call function's handler
- missing test scripts

All of those will be improved later on while review the PR

## Related issue(s)

Not related to any issue

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -5,3 +5,8 @@ build
 typechain-types
 .eslintrc.js
 commitlint.config.js
+
+# subgraph
+subgraph.template.yaml
+generated
+subgraph/scripts
```
Comment (0xmad) [inline .eslintignore:12]: Let's keep new line at the end

File: cli/testScript.sh [hunk 2]
```diff
@@ -11,9 +11,18 @@ node build/ts/index.js setVerifyingKeys \
     --process-messages-zkey ./zkeys/ProcessMessages_10-2-1-2_test/ProcessMessages_10-2-1-2_test.0.zkey \
     --tally-votes-zkey ./zkeys/TallyVotes_10-1-2_test/TallyVotes_10-1-2_test.0.zkey
 node build/ts/index.js create -s 10
+
+# swith to subgraph module
```
Comment (0xmad) [inline cli/testScript.sh:15]:
```suggestion
# switch to subgraph module
```

File: .eslintignore (hunk 1 above)
Comment (0xmad) [inline .eslintignore:12]: Why did you add everything except generated to eslint ignore?

File: cli/ts/utils/constants.ts [hunk 3]
```diff
@@ -8,3 +8,27 @@ export const oldContractAddressStoreName = "contractAddresses.old.json";
 export const contractAddressesStore = path.resolve(__dirname, "..", "..", contractAddressStoreName);
 // local file path where we are storing a previous deployment's contract addresses
 export const oldContractAddressesStore = path.resolve(__dirname, "..", "..", oldContractAddressStoreName);
+// local file name in subgraph module where we are storing the MACI contract address and block number
+export const subgraphNetworkStoreName = "networks.json";
+// local file name in subgraph module where we are storing the previous deployment's MACI contract address and block number
+export const oldSubgraphNetworkStoreName = "networks.old.json";
```
Comment (0xmad) [inline cli/ts/utils/constants.ts:14]: I don't think we need to keep all the previous deploy info.

File: cli/ts/utils/storage.ts [hunk 4]
```diff
@@ -69,3 +69,58 @@ export const doesPathExist = (paths: string[]): [boolean, string | null] => {
...
+  network = "default",
+): void => {
+  // if it does not exist yet, then create it
+  if (!fs.existsSync(subgraphNetworkStore)) {
+    fs.writeFileSync(subgraphNetworkStore, "{}");
+  }
+
+  const networks = readNetworksJSONFile(subgraphNetworkStore);
+
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```
Comment (0xmad) [inline cli/ts/utils/storage.ts:110]: Why did you disable this rule?

File: subgraph/.gitignore [hunk 5]
```diff
@@ -0,0 +1,7 @@
+node_modules
+data
+build
+generated
+
+subgraph.yaml
+bun.lockb
```
Comment (0xmad) [inline subgraph/.gitignore:7]: We don't have `bun.lockb` files and let's keep new line at the end of the file.

File: subgraph/abis/MACI.json
Comment (0xmad) [inline subgraph/abis/MACI.json:1]: All the abis are located in maci-contracts package. It can be installed locally for subgraph package and all the abis should be imported. Right now it's a hardcoded solution which requires to copy-paste files.

File: subgraph/scripts/generateSubgraph.js
Comment (0xmad) [inline subgraph/scripts/generateSubgraph.js:1]: We don't use .js. Use typescript instead.

File: subgraph/scripts/generateSubgraph.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import Mustache from "mustache";
+
+import { readFileSync, writeFileSync } from "fs";
```
Comment (0xmad) [inline subgraph/scripts/generateSubgraph.js:4]:
```suggestion
import fs rom "fs";
```

File: subgraph/scripts/generateSubgraph.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+const network = process.argv.at(2);
+const template = readFileSync("./subgraph.template.yaml", "utf-8");
+// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
```
Comment (0xmad) [inline subgraph/scripts/generateSubgraph.js:8]: Why did you disable eslint for the file?

File: subgraph/src/maci.ts [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+  maci.save();
+
+  // Start indexing the poll; `event.params.pollAddr.poll` is the
+  // address of the new poll contract
+  PollTemplate.create(Address.fromBytes(entity.id));
+}
+
+export function handleOwnershipTransferred(event: OwnershipTransferredEvent): void {
+  const entity = createOrLoadMACI(event);
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
```
Comment (0xmad) [inline subgraph/src/maci.ts:53]: The same here with eslint disabling rules

File: subgraph/src/poll.ts [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+  entity.save();
+
+  const maci = createOrLoadMACI(event);
+  maci.numVotes = maci.numVotes.plus(ONE_BIGINT);
+  maci.save();
+}
+
+export function handleTopupMessage(event: TopupMessageEvent): void {
+  const poll = Poll.load(event.address);
+  if (poll) {
```
Comment (0xmad) [inline subgraph/src/poll.ts:100]:
```suggestion
  const poll = Poll.load(event.address);

  if (poll) {
```

File: subgraph/src/utils/helper.ts [hunk 10]
```diff
@@ -0,0 +1,94 @@
+/* eslint-disable @typescript-eslint/no-inferrable-types */
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable @typescript-eslint/ban-types */
+/* eslint-disable @typescript-eslint/no-shadow */
```
Comment (0xmad) [inline subgraph/src/utils/helper.ts:4]: Please, fix the errors and don't disable linter.
For no-shadow you need to add eslint configuration and add BigInt to exceptions.
no-underscore-dangle - can be left as is.

File: subgraph/subgraph.template.yaml [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+description: Subgraph Indexer for MACI contract
+repository: https://github.com/hangleang/maci-subgraph
+indexerHints:
+  prune: auto
+schema:
+  file: ./schema.graphql
+dataSources:
+  - kind: ethereum
+    name: MACI
+    network: { { network } }
```
Comment (0xmad) [inline subgraph/subgraph.template.yaml:11]:
```suggestion
    network: {{ network }}
```

File: subgraph/subgraph.template.yaml [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    source:
+      address: "{{ address }}"
+      abi: MACI
+      startBlock: { { startBlock } }
```
Comment (0xmad) [inline subgraph/subgraph.template.yaml:15]:
```suggestion
      startBlock: {{ startBlock }}
```

File: subgraph/subgraph.template.yaml [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+          handler: handleDeployPoll
+        - event: OwnershipTransferred(indexed address,indexed address)
+          handler: handleOwnershipTransferred
+        - event: SignUp(uint256,(uint256,uint256),uint256,uint256)
+          handler: handleSignUp
+      file: ./src/maci.ts
+templates:
+  - kind: ethereum
+    name: Poll
+    network: { { network } }
```
Comment (0xmad) [inline subgraph/subgraph.template.yaml:40]:
```suggestion
    network: {{ network }}
```

File: tsconfig.json
Comment (0xmad) [inline tsconfig.json:1]: This file shouldn't be changed

**Review by 0xmad — CHANGES_REQUESTED** [review]
@hangleang thanks, take a look at the maci-contracts package and you can reuse eslint config and tsconfig from there. Left some comments to fix.

File: subgraph/schema.graphql [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+  owner: Bytes! # address
+  stateTreeDepth: Int! # uint8
+  updatedAt: BigInt!
+  blockNumber: BigInt!
+  txHash: Bytes!
+
+  "state"
+  numRegistrar: BigInt!
+  numPoll: BigInt!
+  numVotes: BigInt!
```
Comment (samajammin) [inline subgraph/schema.graphql:12]: `numSignUps`? I believe votes would tracked on `Poll`, not `MAC`

File: cli/ts/utils/constants.ts (hunk 3 above)
Comment (hangleang) [reply cli/ts/utils/constants.ts:14]: okay, so we don't need to backup the previous deployment

File: subgraph/.gitignore (hunk 5 above)
Comment (hangleang) [reply subgraph/.gitignore:7]: noted with thanks

File: cli/ts/utils/storage.ts (hunk 4 above)
Comment (hangleang) [reply cli/ts/utils/storage.ts:110]: resolved! thanks

File: subgraph/scripts/generateSubgraph.js (hunk 6 above)
Comment (hangleang) [reply subgraph/scripts/generateSubgraph.js:4]: removed!

File: subgraph/scripts/generateSubgraph.js (hunk 7 above)
Comment (hangleang) [reply subgraph/scripts/generateSubgraph.js:8]: removed!

File: subgraph/subgraph.template.yaml (hunk 11 above)
Comment (hangleang) [reply subgraph/subgraph.template.yaml:11]: this file will be removed! use the `--network` flag when deploy the subgraph will generate `subgraph.yaml` with corresponding network config from `networks.json` file

File: tsconfig.json
Comment (hangleang) [reply tsconfig.json:1]: has been rolled it back, include the files in `tsconfig.ts` in  `subgraph` module instead

File: subgraph/schema.graphql (hunk 14 above)
Comment (hangleang) [reply subgraph/schema.graphql:12]: thanks, has been changed to `numSignUps`

Comment (samajammin) [reply subgraph/schema.graphql:12]: Has it? I'm still seeing `numVotes`

File: subgraph/schema.graphql [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+  pollId: BigInt! # uint256
+  duration: BigInt! # uint256
+  treeDepth: Int! # uint8
+  maxVoteOptions: BigInt!
+  messageProcessor: Bytes! # address
+  tally: Bytes! # address
+  subsidy: Bytes! # address
+  "merge state after ended"
+  stateRoot: BigInt # uint256
+  numSignups: BigInt! # uint256
```
Comment (samajammin) [inline subgraph/schema.graphql:61]: I believe `numSignups` should be on `MACI` not `Poll`

File: subgraph/.eslintrc.js [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+    requireConfigFile: false,
+    ecmaFeatures: {
+      classes: true,
+      impliedStrict: true,
+    },
+    warnOnUnsupportedTypeScriptVersion: true,
+  },
+  rules: {
+    "no-underscore-dangle": "off",
+    "@typescript-eslint/no-unsafe-assignment": "off",
```
Comment (0xmad) [inline subgraph/.eslintrc.js:22]: It shouldn't be disabled

File: subgraph/schema.graphql (hunk 14 above)
Comment (hangleang) [reply subgraph/schema.graphql]: ohh! my bad, I replied before committing the changes

File: subgraph/.eslintrc.js (hunk 16 above)
Comment (hangleang) [reply subgraph/.eslintrc.js:22]: okay, I'll take a look and fix it later

File: subgraph/schema.graphql (hunk 15 above)
Comment (hangleang) [reply subgraph/schema.graphql:61]: so `numMessages` is number of votes, also including topup messages. is that correct?

Comment (hangleang) [reply subgraph/schema.graphql:61]: regarding the `topup` process, we need which `stateIndex` and how much `voteCredit` to be added to the right state leaf in the tree. but both required params is only available via function call, which need to handle seperately from the event handler, somehow network is not support the tracing API.

**Comment by hangleang** [comment]
### UPDATED
We no longer need to manually generate the `subgraph.yaml` file with `pnpm run codegen:local` command, which is also be removed from the script. it will automatically generate the file before deploying to the graph with specificed `--network` flag

File: .prettierignore [hunk 17]
```diff
@@ -15,3 +15,4 @@ website/.docusaurus/
 website/src/pages/solidity-docs/index.md
 **/deployed-contracts.json
 **/deploy-config.json
+subgraph/generated/
```
Comment (0xmad) [inline .prettierignore:18]: Please add new line at the end

File: subgraph/schema.graphql (hunk 15 above)
Comment (ctrlc03) [reply subgraph/schema.graphql:61]: > w need



> ~I believe `numSignups` should be on `MACI` not `Poll`~ - nevermind, I guess `numSignups` is also on Poll. So @ctrlc03 we now need on `numSignups` on Poll (#1101) in addition to on MACI?

totally missed this. `numSignups` on poll is the number of signups before the poll was "completed" (stateTree and message tree merged)

Comment (ctrlc03) [reply subgraph/schema.graphql:61]: > regarding the `topup` process, we need which `stateIndex` and how much `voteCredit` to be added to the right state leaf in the tree. but both required params is only available via function call, which need to handle seperately from the event handler, somehow network is not support the tracing API.

you have the data here:

https://github.com/privacy-scaling-explorations/maci/blob/dev/contracts/contracts/Poll.sol#L173 

The topup message will be an object:

{

msgType: 2
data: [stateIndex, amount, 0, 0, 0, ...]
}

Comment (hangleang) [reply subgraph/schema.graphql:61]: > > w need
> 
> > ~I believe `numSignups` should be on `MACI` not `Poll`~ - nevermind, I guess `numSignups` is also on Poll. So @ctrlc03 we now need on `numSignups` on Poll (#1101) in addition to on MACI?
> 
> totally missed this. `numSignups` on poll is the number of signups before the poll was "completed" (stateTree and message tree merged)

I'm not quite sure on this, since the users sign up to MACI without specifying to any certain poll. how can we count the number of signup in the certain poll? should we count when user publish message?

Comment (hangleang) [reply subgraph/schema.graphql:61]: > > regarding the `topup` process, we need which `stateIndex` and how much `voteCredit` to be added to the right state leaf in the tree. but both required params is only available via function call, which need to handle seperately from the event handler, somehow network is not support the tracing API.
> 
> you have the data here:
> 
> https://github.com/privacy-scaling-explorations/maci/blob/dev/contracts/contracts/Poll.sol#L173
> 
> The topup message will be an object:
> 
> {
> 
> msgType: 2 data: [stateIndex, amount, 0, 0, 0, ...] }

awesome! I'll update the subgraph based on this

File: .prettierignore (hunk 17 above)
Comment (hangleang) [reply .prettierignore:18]: ohh, thank you

File: subgraph/schema.graphql (hunk 15 above)
Comment (ctrlc03) [reply subgraph/schema.graphql:61]: > > > w need
> > 
> > 
> > > ~I believe `numSignups` should be on `MACI` not `Poll`~ - nevermind, I guess `numSignups` is also on Poll. So @ctrlc03 we now need on `numSignups` on Poll (#1101) in addition to on MACI?
> > 
> > 
> > totally missed this. `numSignups` on poll is the number of signups before the poll was "completed" (stateTree and message tree merged)
> 
> I'm not quite sure on this, since the users sign up to MACI without specifying to any certain poll. how can we count the number of signup in the certain poll? should we count when user publish message?

So you can think of the number of signups to a Poll, the number of users that are signed up up until the end of the poll. So for instance we start 1 poll and 50 users sign up, then we complete it (50 signups there). then we start another poll and 10 more users sign up, then it completed, and we have 60 signups. Does that make sense?

Comment (hangleang) [reply subgraph/schema.graphql:61]: > > regarding the `topup` process, we need which `stateIndex` and how much `voteCredit` to be added to the right state leaf in the tree. but both required params is only available via function call, which need to handle seperately from the event handler, somehow network is not support the tracing API.
> 
> you have the data here:
> 
> https://github.com/privacy-scaling-explorations/maci/blob/dev/contracts/contracts/Poll.sol#L173
> 
> The topup message will be an object:
> 
> {
> 
> msgType: 2 data: [stateIndex, amount, 0, 0, 0, ...] }

thank you for give me some clue, I've updated

anyway, how can I get `stateIndex` and `voteWeight` from the emitted event `PublishMessage` since the message got hashed and encrypted before submitting on-chain?

Comment (hangleang) [reply subgraph/schema.graphql:61]: the reason I want to do this, because I would like to subtract the `voteCredits` from the `voteBalance` state in the certain account linked to the `stateIndex`? or I don’t need to do that, should I store the encrypted data as it is?

Comment (ctrlc03) [reply subgraph/schema.graphql:50]: > from the emitted event `PublishMessage` since the message got hashed and encrypted before submitting on-ch

you're not supposed to get that data, messages are private so they are encrypted before being sent to prevent its content from being seen 🙂

Comment (ctrlc03) [reply subgraph/schema.graphql:50]: > the encry

it'll be fine to store the encrypted data only, thank you!

Comment (hangleang) [reply subgraph/schema.graphql:50]: ahh I see, thank you!

File: subgraph/src/maci.ts [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+  const treeDepths = contract.treeDepths();
+  const durations = contract.getDeployTimeAndDuration();
+
+  entity.pollId = event.params._pollId;
+  entity.messageProcessor = event.params.pollAddr.messageProcessor;
+  entity.tally = event.params.pollAddr.tally;
+  entity.subsidy = event.params.pollAddr.subsidy;
+  entity.maxMessages = maxValues.value0;
+  entity.maxVoteOption = maxValues.value1;
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
```
Comment (0xmad) [inline subgraph/src/maci.ts:31]: Why do you need this disabled rule here?

File: subgraph/src/utils/helper.ts [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+  owner: Bytes | null = null,
+  stateTreeDepth: Int8 = 10,
+): MACI => {
+  let maci = MACI.load(DEFAULT_MACI_ID);
+
+  if (!maci) {
+    maci = new MACI(DEFAULT_MACI_ID);
+    maci.owner = owner !== null ? owner : (event.transaction.from as Bytes);
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    maci.stateTreeDepth = stateTreeDepth as i32;
```
Comment (0xmad) [inline subgraph/src/utils/helper.ts:21]: Why do you need to disable it here and I don't see i32 import in this file.

File: subgraph/src/poll.ts [hunk 20]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+export function handleTopupMessage(event: TopupMessageEvent): void {
+  const stateIndex = event.params._message.data[0].toString();
+  const credits = event.params._message.data[1];
+  const account = Account.load(stateIndex);
+  if (account) {
+    account.voiceCreditBalance = account.voiceCreditBalance.plus(credits);
+    account.save();
+  }
```
Comment (0xmad) [inline subgraph/src/poll.ts:113]:
```suggestion
  const account = Account.load(stateIndex);

  if (account) {
    account.voiceCreditBalance = account.voiceCreditBalance.plus(credits);
    account.save();
  }
```

File: subgraph/src/poll.ts [hunk 21]
```diff
@@ -0,0 +1,132 @@
...
+  entity.poll = event.address;
+  entity.account = stateIndex;
+
+  entity.blockNumber = event.block.number;
+  entity.timestamp = event.block.timestamp;
+  entity.txHash = event.transaction.hash;
+  entity.save();
+
+  const poll = Poll.load(event.address);
+  if (poll) {
```
Comment (0xmad) [inline subgraph/src/poll.ts:127]:
```suggestion
  const poll = Poll.load(event.address);

  if (poll) {
```

File: subgraph/src/maci.ts [hunk 22]
```diff
@@ -0,0 +1,82 @@
...
+  const user = createOrLoadUser(event.params._userPubKeyX, event.params._userPubKeyY, event);
+  createOrLoadAccount(event.params._stateIndex, event, user.id, event.params._voiceCreditBalance);
+
+  const maci = createOrLoadMACI(event);
+  maci.numSignUps = maci.numSignUps.plus(ONE_BIGINT);
+  maci.updatedAt = event.block.timestamp;
+  maci.save();
+
+  const poll = Poll.load(maci.latestPoll);
+  if (poll) {
```
Comment (0xmad) [inline subgraph/src/maci.ts:77]:
```suggestion
  const poll = Poll.load(maci.latestPoll);

  if (poll) {
```

File: subgraph/src/poll.ts [hunk 23]
```diff
@@ -0,0 +1,132 @@
...
+  entity.data = event.params._message.data;
+  entity.poll = event.address;
+
+  entity.blockNumber = event.block.number;
+  entity.timestamp = event.block.timestamp;
+  entity.txHash = event.transaction.hash;
+  entity.save();
+
+  const poll = Poll.load(event.address);
+  if (poll) {
```
Comment (0xmad) [inline subgraph/src/poll.ts:99]:
```suggestion
  const poll = Poll.load(event.address);

  if (poll) {
```

File: subgraph/src/utils/helper.ts (hunk 19 above)
Comment (hangleang) [reply subgraph/src/utils/helper.ts:21]: It is required for typecasting in AssemlyScript, but I will change BigInt, cast the type instead

File: subgraph/src/maci.ts (hunk 18 above)
Comment (hangleang) [reply subgraph/src/maci.ts:31]: I could not figuring out a way to overcome this...

Comment (hangleang) [reply subgraph/src/maci.ts:31]: I could not figuring out a way to overcome this...

**Comment by hangleang** [comment]
sorry, click on the wrong button, please help remove the review request

File: subgraph/.eslintrc.js [hunk 24]
```diff
@@ -0,0 +1,35 @@
+const path = require("path");
+
+module.exports = {
+  root: true,
+  extends: ["../.eslintrc.js", "plugin:@typescript-eslint/recommended"],
+  parser: "@typescript-eslint/parser",
+  plugins: ["@typescript-eslint"],
```
Comment (0xmad) [inline subgraph/.eslintrc.js:7]: Why did you add `@typescript-eslint` here? This one is added for root eslint conifg.

File: subgraph/src/maci.ts (hunk 18 above)
Comment (0xmad) [reply subgraph/src/maci.ts:31]: So, if it was something related to eslint, it will be a problem for maxValues as well, so seems it's a type mismatch error. You need to check `treeDepths` and and `treeDepth` for entity.

File: subgraph/src/utils/helper.ts [hunk 25]
```diff
@@ -0,0 +1,70 @@
...
+  event: ethereum.Event,
+  owner: Bytes | null = null,
+  stateTreeDepth: GraphBN = GraphBN.fromI32(10),
+): MACI => {
+  let maci = MACI.load(DEFAULT_MACI_ID);
+
+  if (!maci) {
+    maci = new MACI(DEFAULT_MACI_ID);
+    maci.owner = owner !== null ? owner : (event.transaction.from as Bytes);
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
```
Comment (0xmad) [inline subgraph/src/utils/helper.ts:20]: Here is the same issue, maybe it can be cast as the line above (with as Bytes but here with as I32)

File: subgraph/.eslintrc.js (hunk 24 above)
Comment (hangleang) [reply subgraph/.eslintrc.js]: It has been removed!

File: subgraph/src/utils/helper.ts (hunk 25 above)
Comment (hangleang) [reply subgraph/src/utils/helper.ts]: I've changed those to bigint for easier work around, thank you

File: cli/deploy-config.json
Comment (0xmad) [inline cli/deploy-config.json:1]: Please remove this file and deployed-contracts.json

**Review by 0xmad — APPROVED** [review]
@hangleang thanks!

**Comment by ctrlc03** [comment]
@hangleang could you rebase and fix the pnpm-lock conflict please?

**Review by 0xmad — APPROVED** [review]
@hangleang thanks, can you squash your commits to fix commitlint check and describe in commit message general work done for this task?
